app/townsquare/views.py

```diff
@@ -332,6 +332,18 @@ def get_following_tribes(request):
 
 
 def town_square(request):
+    from app.redis_service import RedisService
+    redis = RedisService().redis
```

Comment:
The redis instance needs to be instantiated globally - connecting it here on every request is going to cause performance issues in production.

Reply:
I created one instance to module level and modify the creation of Redis instances, now it uses a singleton pattern.